### PR TITLE
DLUHC-200 Add base Timetable component

### PR DESCRIPTION
**Pull request by amjones-scottlogic**
This PR adds the base Timetable component.
- Adds the Timetable Page
- Adds default values and html for Timetable Page
- Adds the base Timetable component
- Adds Tanstack Table to manage the Timetable data
- Adds the base Progress Tile
- Updates the index to display the Timetable Page (this isnt an optimal solution for when we work on other pages / components)

There is still a lot of work to do on this Page, primarily styling, loading all the data and expanding the more info sections. These will be done in separate PRs. Anything else left to do I should have added a comment for.

**Screenshot of Timetable Page**

image

### CODE REVIEW

File: dluhc-component-library/src/components/timetablePage/ProgressTile.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+interface ProgressTileProps {
+  progress: "notStarted" | "delayed" | "inProgress" | "finished";
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/components/timetablePage/ProgressTile.tsx:2]: This type should be pulled out into a type file and reused in the Timetable component

File: dluhc-component-library/src/components/timetablePage/ProgressTile.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+const ProgressTile = ({ progress }: ProgressTileProps) => {
+  const progressClass = `progress-${progress}`;
+
+  return (
+    <span className={`uppercase font-semibold py-1 px-2 ${progressClass}`}>
+      {progress}
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/components/timetablePage/ProgressTile.tsx:10]: This value currently looks like "notStarted" so it needs to be converted to a human readable string

File: dluhc-component-library/src/components/timetablePage/Timetable.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+type RowData = {
+  developmentPlanEvent: string;
+  startDate: string;
+  endDate: string;
+  progress: "notStarted" | "delayed" | "inProgress" | "finished";
+};
+
+const columnHelper = createColumnHelper<RowData>();
+
+const columns = [
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/components/timetablePage/Timetable.tsx:20]: Column definitions should be moved out into their own file

File: dluhc-component-library/src/index.css [hunk 4]
```diff
@@ -1,3 +1,7 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
+body {
+  font-family: "GDS Transport", arial, sans-serif;
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/index.css:6]: The GDS Transport font doesnt currently work as its not included. This will be fixed in another PR

File: dluhc-component-library/src/main.tsx [hunk 5]
```diff
@@ -1,24 +1,36 @@
 import { render } from "preact";
 import SiteSelectionForm from "./components/siteSelectionForm.tsx";
+import TimetablePage from "./components/timetablePage/TimetablePage.tsx";
 
 import "./index.css";
 
-type renderFunction = (options: {}, element: HTMLElement) => void;
+type Options = any;
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/main.tsx:7]: Some thought needs to go into this type so I have pulled it out to at least name it for now.

File: dluhc-component-library/src/mockData/timetable.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
+[
+  {
+    "developmentPlanEvent": "Scoping of the New Local Plan",
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/mockData/timetable.json:3]: The actual format, keys and values of this file are not final and may be updated in the future

File: dluhc-component-library/src/components/timetablePage/TimetablePage.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+interface TimetablePageProps {
+  filepath: string;
+}
+
+const loadData = async (filepath: string) =>
+  await fetch(filepath).then((res) => res.json());
+
+const TimetablePage = ({ filepath }: TimetablePageProps) => {
+  const [timetableData, setTimetableData] = useState();
```
Comment (samornsby) [inline dluhc-component-library/src/components/timetablePage/TimetablePage.tsx:12]: Could we pull `RowData` out into a new file so we can use it for the state type here too?

Comment (amjones-scottlogic) [reply dluhc-component-library/src/components/timetablePage/TimetablePage.tsx:12]: Yeah ill do that when i pull the progress type out too
